charon/src/ast/types.rs

```diff
@@ -390,13 +390,18 @@ pub struct Layout {
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Drive, DriveMut)]
 pub struct VTable;
 
-/// The kind of meta information associated with a Dynamic Sized Type (DST).
-/// For `str` and slices like `[u8]`, this is the length of the slice.
-/// For trait objects, this is the vtable of the trait.
-/// For user defined DSTs, this is inductively inherited from their last field.
+/// The metadata stored in a pointer. That's the information stored in pointers alongside
+/// their address. It's empty for `Sized` types, and interesting for unsized
+/// aka dynamically-sized types.
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Drive, DriveMut)]
-pub enum DstMetaKind {
+pub enum PtrMetadata {
+    /// Types that need no metadata, namely `T: Sized` types.
+    NoMetadata,
```

Comment:
In Rust we try not to repeat the type name in variant names. To avoid ocaml clashes we can rename individual variants like so:
```suggestion
    #[charon::rename("NoMetadata")]
    None,
```